### PR TITLE
config: add Shutdown method

### PR DESCRIPTION
**Pull request by codeboten**
This allows users to call Shutdown on the tracer/meter providers configured in the SDK.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -41,9 +52,18 @@ func NewSDK(opts ...ConfigurationOption) (SDK, error) {
...
 	return SDK{
-		meterProvider:  initMeterProvider(o),
-		tracerProvider: initTracerProvider(o),
+		meterProvider:  mp,
+		tracerProvider: tp,
+		shutdown: func(ctx context.Context) error {
+			// TODO: handle errors
+			mpShutdown(ctx)
+			tpShutdown(ctx)
+			return nil
```
Comment (pellared) [inline config/config.go:65]: how about

```suggestion
			err : = mp.Shutdown(ctx)
			return errors.Join(tp.Shutdown(ctx))
```

File: config/config.go [hunk 2]
```diff
@@ -41,9 +52,18 @@ func NewSDK(opts ...ConfigurationOption) (SDK, error) {
 		o = opt.apply(o)
 	}
 
+	mp, mpShutdown := initMeterProvider(o)
```
Comment (pellared) [inline config/config.go:55]: I think you can considering looking at this code https://github.com/open-telemetry/opentelemetry-go/blob/main/example/dice/otel.go as reference for `NewSDK`.

File: config/config.go (hunk 1 above)
Comment (codeboten) [reply config/config.go]: updated

**Review by pellared — APPROVED** [review]
It might be good to add a changelog entry.

**Comment by codeboten** [comment]
Thanks for the review @pellared, changelog added

**Review by hanyuancheung — APPROVED** [review]

**Pull request merged**